ibm/resource_ibm_resource_tag.go

```diff
@@ -0,0 +1,312 @@
+			MinValueLength:             1,
+			MaxValueLength:             128})
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 tags,
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^[A-Za-z0-9:_ .-]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             128})
+
+	ibmResourceTagValidator := ResourceValidator{ResourceName: "ibm_resource_tag", Schema: validateSchema}
+	return &ibmResourceTagValidator
+}
+
+func resourceIBMResourceTagCreate(d *schema.ResourceData, meta interface{}) error {
+	var rType, tType string
+	resources := []globaltaggingv1.Resource{}
+
+	userDetails, err := meta.(ClientSession).BluemixUserDetails()
+	if err != nil {
+		return err
+	}
+	accountID := userDetails.userAccount
+
+	gtClient, err := meta.(ClientSession).GlobalTaggingAPIv1()
+	if err != nil {
+		return fmt.Errorf("Error getting global tagging client settings: %s", err)
+	}
+
+	resourceID := d.Get(resourceID).(string)
+	if v, ok := d.GetOk(resourceType); ok && v != nil {
+		rType = v.(string)
+	}
+
+	r := globaltaggingv1.Resource{ResourceID: ptrToString(resourceID), ResourceType: ptrToString(rType)}
+	resources = append(resources, r)
+
+	var add []string
+	if v, ok := d.GetOk(tags); ok {
+		tags := v.(*schema.Set)
+		for _, t := range tags.List() {
+			add = append(add, fmt.Sprint(t))
+		}
+	}
+
+	schematicTags := os.Getenv("IC_ENV_TAGS")
+	var envTags []string
+	if schematicTags != "" {
+		envTags = strings.Split(schematicTags, ",")
+		add = append(add, envTags...)
+	}
+
+	AttachTagOptions := &globaltaggingv1.AttachTagOptions{}
+	AttachTagOptions.Resources = resources
+	AttachTagOptions.TagNames = add
+	if v, ok := d.GetOk(tagType); ok && v != nil {
+		tType = v.(string)
+		AttachTagOptions.TagType = ptrToString(tType)
+
+		if tType == service {
+			AttachTagOptions.AccountID = ptrToString(accountID)
+		}
+	}
+
+	if len(add) > 0 {
+		_, resp, err := gtClient.AttachTag(AttachTagOptions)
+		if err != nil {
+			return fmt.Errorf("Error attaching resource tags >>>>  %v : %s", resp, err)
+		}
+	}
+
+	crn, err := regexp.Compile(crnRegex)
+	if err != nil {
+		return err
+	}
+
+	if crn.MatchString(resourceID) {
+		d.SetId(resourceID)
+	} else {
+		d.SetId(fmt.Sprintf("%s/%s", resourceID, resourceType))
+	}
+
+	return resourceIBMResourceTagRead(d, meta)
+}
+
+func resourceIBMResourceTagRead(d *schema.ResourceData, meta interface{}) error {
+	var rID, rType, tType string
+
+	userDetails, err := meta.(ClientSession).BluemixUserDetails()
+	if err != nil {
+		return err
+	}
+	acctID := userDetails.userAccount
+
+	crn, err := regexp.Compile(crnRegex)
+	if err != nil {
+		return err
+	}
+
+	if crn.MatchString(d.Id()) {
+		rID = d.Id()
+	} else {
+		parts, err := vmIdParts(d.Id())
+		if err != nil {
+			return err
+		}
+		rID = parts[0]
+		rType = parts[1]
+	}
+
+	if v, ok := d.GetOk(tagType); ok && v != nil {
```

Comment:
API is not giving back which type the tag it is?

Reply:
API doesn't give value for tag_type